### ISSUE TITLE
fix: Audio clicks/pops when adjusting Freeverb filter parameters and changing waveform frequency

### ISSUE THREAD

**Issue by sbauly** [alnitak/flutter_soloud#156]
**Description**

Audible artifacts (clicks and pops) can be heard when modifying audio parameters in real-time. This is particularly noticeable with Freeverb's wetness parameter, but can also be heard when adjusting a generated waveform's frequency.

**Steps To Reproduce**

1. Adjust the frequency slider in the [waveform example,](https://marcobavagnoli.com/flutter_soloud_waveform/) or play any sound with the Freeverb filter and adjust the wetness during playback.
2. Hear the clicks and pops in audio as the frequency or wetness is adjusted
3. Many pops can be heard when dragging a slider (rapidly adjusting a parameter), but singular pops can also be heard when tapping just once to adjust a parameter. This is especially noticeable whilst wearing headphones.

**Recommendations**

It would be great if we could implement smoothing/interpolation when changing filter values to eliminate these artifacts. This would allow for seamless transitions, which is particularly important for immersion in games.

**Comment by alnitak** [comment]
Hi @sbauly,

I can hear clicks when fast changing the frequency in the *waveform example*, but not when changing parameters like the *wet* in the freeverb filter.

I tried on linux, android and [web](https://marcobavagnoli.com/flutter_soloud_example/).

I'll try to figure out what wrong. 
Thank you.

**Comment by sbauly** [comment]
Hi @alnitak, thank you for your reply!

I threw together a quick example app to demonstrate the issue. 

Here is the result of adjusting the wetness parameter on generated waveforms, with audible clicks and pops when adjusting the wetness:


https://github.com/user-attachments/assets/edf6d342-bb3e-4262-a022-82da35845c12

Here is the result of adjusting the wetness parameter on an audio file (I tested with .ogg, .wav and .mp3, all were producing pops and clicks):


https://github.com/user-attachments/assets/7592d5b5-eaf1-4b2b-b3cd-87854dfc03f4

The pops are quieter when adjusting the _wet_ on an audio file, but still noticeable with headphones.

I have been adjusting the wetness as follows:

```
Center(
              child: Slider(
                value: viewModel.reverbWet,
                onChanged: (value) {
                  viewModel.adjustFreeverbWetness(value);
                },
              ),
            ),
```

```
  double _reverbWet = 0.5;
  double get reverbWet => _reverbWet;

  void adjustFreeverbWetness(double wetness) {
    _reverbWet = wetness;
    SoLoud.instance.filters.freeverbFilter.wet.value = wetness;
    notifyListeners();
  }
```

I've tried using the `fadeFilterParameter()` method instead of directly setting the wetness value, but have been unable to get it to work yet.

I'm using `flutter_soloud: ^2.1.7` from pub.dev, and I can hear these pops on both Android and iOS.

As always, thank you for your amazing hard work on this package!

**Comment by alnitak** [comment]
Hey @sbauly,

It was hard, but I think I fixed the waveform clicks and pops. 
I didn't find what was wrong about the clicks with the wetness on freeverb. I'll try again.

The #158 has already been merged if you want to give it a try.

**Comment by sbauly** [comment]
Thank you very much @alnitak, I just tested this and the frequencies are changing flawlessly and smoothly!

Regarding Freeverb, I've noticed that increasing the wetness also increases the volume/gain quite dramatically. This is fairly normal behavior with reverb, but is more pronounced when the input is already quite loud. The faster the wetness is adjusted, the more clicks and pops are heard.

I suspect that this is the cause of the pops. Again, this might be something the new limiter and compressor filters in #149 can hopefully help alleviate :)
